src/table.rs

```diff
@@ -56,6 +56,11 @@ impl Table {
 
         let metadata = self.read_table_metadata(&path).await?;
         // TODO: check if the metadata is out of date.
+        if metadata.last_updated_ms == 0 {
+            return Err(anyhow!(
```

Comment:
We need to return `icelake::Error` instead.

Reply:
The whole table.rs use `anyhow::Result` now. We can create a new pr to switch it to `icelake::Result`.

Reply:
Great!